src/hotspot/share/runtime/os.cpp

```diff
@@ -1045,16 +1045,11 @@ void os::print_hex_dump(outputStream* st, const_address start, const_address end
   while (p < end) {
     if (cols == 0) {
       // highlight start of line if address of interest is located there
-      if (highlight_address != nullptr && highlight_address == p) {
-        st->print("=>" PTR_FORMAT ":   ", p2i(logical_p));
-      } else {
-        // keep for now old formatting intact
-        if (highlight_address != nullptr) {
-          st->print("  " PTR_FORMAT ":   ", p2i(logical_p));
-        } else {
-          st->print(PTR_FORMAT ":   ", p2i(logical_p));
-        }
-      }
+      bool should_highlight = false;
+      if (highlight_address == p) should_highlight = true;
```

Comment:
Would this not still be restricted to highlighting if the address matches the start of the line? Would you not need something like this instead (not tested):

```
if (highlight_address >= p && highlight_address < p + cols_per_line/unitsize)
```

Reply:
Hi Thomas, the coding is now

`if (highlight_address >= p && highlight_address < p + cols_per_line) should_highlight = true;`

Seems you check an old revision ?

And the gtest tests now also  `from+60`  which is not at a line start .